files/common/usr/bin/migrationctl

```diff
@@ -0,0 +1,282 @@
+            return color + text + Color.RESET
+        return text
+
+    @staticmethod
+    def red(text):
+        return Color.colorize(text, Color.RED)
+
+    @staticmethod
+    def green(text):
+        return Color.colorize(text, Color.GREEN)
+
+    @staticmethod
+    def yellow(text):
+        return Color.colorize(text, Color.YELLOW)
+
+    @staticmethod
+    def normal(text):
+        return text
+
+class Service:
+    NAME_HEADER_LEN = 40
+
+    def __init__(self, name):
+        self.name = name
+        self.state = None
+        self.refresh()
+
+    @staticmethod
+    def print_header():
+        print('%-*s %s' % (Service.NAME_HEADER_LEN, 'SERVICE', 'STATE'))
+
+    def __str__(self):
+        return self.name
+
+    def refresh(self):
+        self.state = sh.systemctl('show', '--no-page', '--value', '-p',
+                                  'ActiveState', self.name).strip()
+
+
+    def _command(self, command):
+        try:
+            sh.systemctl(command, self.name)
+            self.refresh()
+            return True
+        except sh.ErrorReturnCode:
+            return False
+
+    def start(self):
+        started = self._command('start')
+        if started and self.state != 'active':
+            warning('Service %s start succeeded but state is %s'
+                    % (self.name, self.state))
+            return False
+
+        return started
+
+    def stop(self):
+        return self._command('stop')
+
+    def restart(self):
+        return self._command('restart')
+
+    def enable(self):
+        return self._command('enable')
+
+    def disable(self):
+        return self._command('disable')
+
+    def is_active(self):
+        return self.state == 'active'
+
+    def is_inactive(self):
+        return self.state == 'inactive'
+
+    def is_failed(self):
+        return self.state == 'failed'
+
+    def is_activating(self):
+        return self.state == 'activating'
+
+    def is_deactivating(self):
+        return self.state == 'deactivating'
+
+    def pretty_print_state(self):
+        STATES = {
+            'active': Color.green,
+            'inactive': Color.normal,
+            'activating': Color.yellow,
+            'deactivating': Color.yellow,
+            'failed': Color.red
+        }
+
+        colorize = STATES.get(self.state, Color.red)
+        print('%-*s %s' % (Service.NAME_HEADER_LEN, self.name,
+                           colorize(self.state)))
+
+class Migration:
+    PERFORM_MIGRATION = '/var/delphix/migration/perform-migration'
+    MIGRATION_CHECK_DIR = '/var/delphix/migration/illumos-config'
+
+    def __init__(self):
+        Migration.system_check()
+
+        self.svc_migration = Service('delphix-migration.service')
+        self.svc_iscsi = Service('rtslib-fb-targetctl.service')
+        self.svc_platform = Service('delphix-platform.service')
+        self.svc_postgres = Service('delphix-postgres@default.service')
+        self.svc_mgmt = Service('delphix-mgmt.service')
```

Comment:
I haven't really given this script a close look.. but why don't we have our other services listed here? e.g. `delphix-stat`, `delphix-fluentd`, `delphix-nginx`, etc.. how did you come to include these services, and exclude others? What about the NFS services? and/or any other system services that may be relevant (e.g. perhaps ssh)?

Reply:
I've selected the services that are affected by the migration code. I haven't really looked at the other services, but it would probably be a good idea to start them as well if they have failed due to missing dependencies on delphix-platform.

I think once the changes for `delphix.target` are in we can also start that service to make sure that all of the Delphix services are started successfully.

Reply:
> I think once the changes for delphix.target are in we can also start that service to make sure that all of the Delphix services are started successfully.

`delphix.target` is in the latest trunk builds.. I think we'll need to restart/start this target, since otherwise, we'll fail to restart/start all the other services that comprise the appliance.. as-is, I think they'd fail due to `delphix-platform.service` being down, and then we'd never bring the back up. Have you tested this at all?

Reply:
Yes, I've tested this for all the other service. I've rebased my code on top of the delphix.target changes and will be able to test it soon. Let me push my new changes (not ethat I will change the commit history later).